src/stdlib_logger.f90

```diff
@@ -567,6 +568,31 @@ pure subroutine configure( self, add_line, indent, max_width, time_stamp )
     end subroutine configure
 
 
+    subroutine final_logger( self )
+!! finalizes the logger_t entity by flushing the units
+        type(logger_t), intent(in) :: self
+
+        integer        :: iostat
+        character(256) :: message
+        integer        :: unit
+
+        do unit=1, self % units
+            flush( self % log_units(unit), iomsg=message, iostat=iostat )
+            if ( iostat /= 0 ) then
+                write(error_unit, '(a, i0)' ) 'In the logger_t finalizer ' // &
+                    'an error occured in flushing UNIT = ',                   &
+                    self % log_units(unit)
+                write(error_unit, '(a, i0)') 'With IOSTAT = ', iostat
+                write(error_unit, '(a)') 'With IOMSG = ' // trim(message)
+
+            end if
+
+        end do
+
+        return
```

Comment:
`return` could be deleted, right?

Reply:
I agree, and we should probably mention this in the style guide as well.